### PR TITLE
Restructure lang support

### PR DESCRIPTION
**Pull request by macovedj**
Implements changes described [here](https://github.com/bytecodealliance/component-docs/issues/43)

### CODE REVIEW

File: component-model/src/language-support/javascript.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+Now, use `jco` to create a component from the JS module:
+
+```sh
+$ jco componentize add.js --wit add.wit -n example -o add.wasm
+OK Successfully written add.wasm with imports ().
+```
+
+Now, run the component using the Rust `add` host:
```
Comment (itowlson) [inline component-model/src/language-support/javascript.md:31]: This probably needs a link or other pointer to the Add host (previously the Add host was mentioned earlier on the same page but now that's split away).

File: component-model/src/language-support/python.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+[Install `componentize-py`](https://github.com/dicej/componentize-py#installing-from-pypi) and
+generate a component from `guest.py`.
+
+```sh
+$ pip install componentize-py
+$ componentize-py -d ../examples/example-host/add.wit -w example componentize guest -o add.wasm
+Component built successfully
+```
+
+To test the component, run it using the Rust `add` host:
```
Comment (itowlson) [inline component-model/src/language-support/python.md:31]: Again maybe add a link?

**Review by itowlson — COMMENTED** [review]
LGTM except two small things that now need linking.  Thanks!

File: component-model/src/language-support/rust.md [hunk 3]
```diff
@@ -13,7 +13,70 @@ cargo install --git https://github.com/bytecodealliance/cargo-component --locked
 
 > There is currently no binary or `crates.io` distribution of `cargo component`.
 
-## Creating a library component with `cargo component`
+## Building a Component with `cargo component`
+
+[`cargo-component`](https://github.com/bytecodealliance/cargo-component) is a `cargo` subcommand for
+creating WebAssembly components using Rust as the component's implementation language.
+
+Let's create the same `add` component using the `cargo-component` tooling. First scaffold a project:
```
Comment (kate-goldenring) [inline component-model/src/language-support/rust.md:21]: Someone could land on this page before the `language-support.md` page. Should we add a section to that page saying "this is the add component -- our prevailing example"?

File: component-model/src/language-support/rust.md [hunk 4]
```diff
@@ -13,7 +13,70 @@ cargo install --git https://github.com/bytecodealliance/cargo-component --locked
 
 > There is currently no binary or `crates.io` distribution of `cargo component`.
 
-## Creating a library component with `cargo component`
+## Building a Component with `cargo component`
+
+[`cargo-component`](https://github.com/bytecodealliance/cargo-component) is a `cargo` subcommand for
+creating WebAssembly components using Rust as the component's implementation language.
```
Comment (kate-goldenring) [inline component-model/src/language-support/rust.md:19]: This is redundant with the paragraph a few paragraphs above. I think we can remove this.
```suggestion
```

**Review by kate-goldenring — COMMENTED** [review]
My comments echo Ivans. Just a couple small changes.

File: component-model/src/language-support/javascript.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+A component can be created from a JS module using `jco componentize`. First, install `jco` and
+`componentize-js`:
+
+```sh
+$ npm install @bytecodealliance/jco
+$ npm install @bytecodealliance/componentize-js
+```
+
+Create a JavaScript module that implements the `add` function in [`add.wit`](../examples/example-host/add.wit):
```
Comment (kate-goldenring) [inline component-model/src/language-support/javascript.md]: I think this link (and others that reference this WIT in this PR) needs to be one parent directory higher. If you use VSCode, you could run this extension on each file you've changed to make sure the relative links are adjusted: https://marketplace.visualstudio.com/items?itemName=blackmist.LinkCheckMD

```suggestion
Create a JavaScript module that implements the `add` function in [`add.wit`](../../examples/example-host/add.wit):
```

File: component-model/src/language-support/python.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
+# Python Tooling
+
+### Building a Component with `componentize-py`
+
+[`componentize-py`](https://github.com/dicej/componentize-py) is a tool that converts a Python
+application to a WebAssembly component.
+
+Create a Python program that implements the `add` function in the [`example`
+world](../examples/example-host/add.wit). Note that it imports the bindings that will be created by
```
Comment (kate-goldenring) [inline component-model/src/language-support/python.md]:
```suggestion
world](../../examples/example-host/add.wit). Note that it imports the bindings that will be created by
```

File: component-model/src/language-support/python.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        return x + y
+EOT
+```
+
+[Install `componentize-py`](https://github.com/dicej/componentize-py#installing-from-pypi) and
+generate a component from `guest.py`.
+
+```sh
+$ pip install componentize-py
+$ componentize-py -d ../examples/example-host/add.wit -w example componentize guest -o add.wasm
```
Comment (kate-goldenring) [inline component-model/src/language-support/python.md:27]:
```suggestion
$ componentize-py -d /path/to/examples/example-host/add.wit -w example componentize guest -o add.wasm
```

File: component-model/src/language-support/rust.md [hunk 8]
```diff
@@ -13,7 +13,69 @@ cargo install --git https://github.com/bytecodealliance/cargo-component --locked
 
 > There is currently no binary or `crates.io` distribution of `cargo component`.
 
-## Creating a library component with `cargo component`
+## Building a Component with `cargo component`
+
+Create a Rust program that implements the `add` function in the [`example`
+world](../examples/example-host/add.wit). Note that it imports the bindings that will be created by
```
Comment (kate-goldenring) [inline component-model/src/language-support/rust.md]:
```suggestion
world](../../examples/example-host/add.wit). Note that it imports the bindings that will be created by
```

**Pull request merged**